sdk-java/src/test/java/io/littlehorse/sdk/wfsdk/internal/WorkflowThreadImplTest.java

```diff
@@ -107,4 +107,65 @@ void testWhileLoopConditional() {
 
         assertThat(lastNodeInLoopBody.getOutgoingEdgesCount()).isEqualTo(2);
     }
+
+    @Test
+    void noRetriesByDefault() {
+        WorkflowImpl wf = new WorkflowImpl("asdf", thread -> {
+            thread.execute("asdf");
+        });
+        PutWfSpecRequest wfSpec = wf.compileWorkflow();
+        Node taskNode =
+                wfSpec.getThreadSpecsOrThrow(wfSpec.getEntrypointThreadName()).getNodesOrThrow("1-asdf-TASK");
+
+        assertThat(taskNode.getTask().getRetries()).isEqualTo(0);
+    }
+
+    @Test
+    void setIndividualRetry() {
+        WorkflowImpl wf = new WorkflowImpl("asdf", thread -> {
+            thread.execute("asdf").withRetries(2);
+            ;
+        });
+        PutWfSpecRequest wfSpec = wf.compileWorkflow();
+        Node taskNode =
+                wfSpec.getThreadSpecsOrThrow(wfSpec.getEntrypointThreadName()).getNodesOrThrow("1-asdf-TASK");
+
+        assertThat(taskNode.getTask().getRetries()).isEqualTo(2);
+    }
+
+    @Test
+    void setDefaultRetryAndOverride() {
+        WorkflowImpl wf = new WorkflowImpl("asdf", thread -> {
+            thread.execute("asdf");
+            thread.execute("asdf").withRetries(2);
+        });
+        wf.setDefaultTaskRetries(1);
+        PutWfSpecRequest wfSpec = wf.compileWorkflow();
+        Node defaultNode =
+                wfSpec.getThreadSpecsOrThrow(wfSpec.getEntrypointThreadName()).getNodesOrThrow("1-asdf-TASK");
+
+        assertThat(defaultNode.getTask().getRetries()).isEqualTo(1);
+
+        Node overridenNode =
+                wfSpec.getThreadSpecsOrThrow(wfSpec.getEntrypointThreadName()).getNodesOrThrow("2-asdf-TASK");
+        assertThat(overridenNode.getTask().getRetries()).isEqualTo(2);
+    }
+
+    @Test
+    void setDefaultTaskTimeout() {
+        WorkflowImpl wf = new WorkflowImpl("asdf", thread -> {
```

Comment:
I think it is worth to have more real-life names for the variables we have on the test scenarios that makes it more clear the intention of the test and also helps to debug easier if needed